c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1439 @@
+    uint : 24;
+} timer_control_t;
+
+//! Values for ::timer_control_t::pre_divide
+enum timer_pre_divide {
+    TIMER_PRE_DIVIDE_1 = 0,     //!< Divide by 1
+    TIMER_PRE_DIVIDE_16 = 1,    //!< Divide by 16
+    TIMER_PRE_DIVIDE_256 = 2    //!< Divide by 256
+};
+
+//! Timer interrupt status flag
+typedef struct {
+    uint status : 1;            //!< The flag bit
+    uint : 31;
+} timer_interrupt_status_t;
+
+//! Timer controller registers
+typedef struct {
+    uint load_value;            //!< Load value for Timer
+    const uint current_value;   //!< Current value of Timer
+    timer_control_t control;    //!< Timer control register
+    uint interrupt_clear;       //!< Interrupt clear (any value may be written)
+    //! Timer raw interrupt status
+    const timer_interrupt_status_t raw_interrupt_status;
+    //! Timer masked interrupt status
+    const timer_interrupt_status_t masked_interrupt_status;
+    uint background_load_value; //!< Background load value for Timer
+    uint _dummy;
+} timer_controller_t;
+
+ASSERT_WORD_SIZED(timer_control_t);
+ASSERT_WORD_SIZED(timer_interrupt_status_t);
+
+//! Timer 1 control registers
+static volatile timer_controller_t *const timer1_control =
+        (timer_controller_t *) TIMER1_BASE;
+//! Timer 2 control registers
+static volatile timer_controller_t *const timer2_control =
+        (timer_controller_t *) TIMER2_BASE;
+
+//! \}
+
+// ---------------------------------------------------------------------
+//! \name 7. DMA Controller
+//! \brief Each ARM968 processing subsystem includes a DMA controller.
+//! \details The DMA controller is primarily used for transferring inter-neural
+//!     connection data from the SDRAM in large blocks in response to an input
+//!     event arriving at a fascicle processor, and for returning updated
+//!     connection data during learning. In addition, the DMA controller can
+//!     transfer data to/from other targets on the System NoC such as the
+//!     System RAM and Boot ROM.
+//!
+//!     As a secondary function the DMA controller incorporates a ‘Bridge’
+//!     across which its host ARM968 has direct read and write access to
+//!     System NoC devices, including the SDRAM. The ARM968 can use the
+//!     Bridge whether or not DMA transfers are active.
+//! \{
+
+//! DMA descriptor
+typedef struct {
+    uint _zeroes : 2;       //!< Must be zero
+    uint length_words : 15; //!< length of the DMA transfer, in words
+    uint : 2;
+    //! read from or write to system bus, see ::dma_direction_t
+    uint direction : 1;
+    uint crc : 1;           //!< check (read) or generate (write) CRC
+    //! burst length = 2<sup>B</sup>&times;Width, B = 0..4 (i.e max 16)
+    uint burst : 3;
+    uint width : 1;         //!< transfer width, see ::dma_transfer_unit_t
+    uint privilege : 1;     //!< DMA transfer mode is user (0) or privileged (1)
+    uint transfer_id : 6;   //!< software defined transfer ID
```

Comment:
why 6?

Reply:
See page 24 of the datasheet.
![image](https://user-images.githubusercontent.com/1155878/86768737-e83fe780-c045-11ea-84d7-4e7e8737a2a1.png)
It's six because the datasheet says it is six.